### PR TITLE
connect-frontend-backend

### PR DESCRIPTION
**Pull request by aviadch**

### CODE REVIEW

File: frontend/src/screens/WaitingRoom.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import WaitingPlayers from "../components/WaitingPlayers";
+import Button from "@material-ui/core/Button";
+import { MyContext } from "../infra/MyContext.js";
+
+const MOCK_playerList = ["Elisha", "Lahav", "Aviad"];
+class WaitingRoom extends Component {
+  constructor() {
+    super();
+    this.state = {
+      isUserEntered: false,
```
Comment (AdiYElisha) [inline frontend/src/screens/WaitingRoom.js:13]: do you really have to have this bool?

File: frontend/src/screens/WaitingRoom.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                {!context.state.isUserEntered ? (
+                  <>
+                    <EnterName />
+                  </>
+                ) : (
+                  <></>
+                )}
+
+                <Button variant="contained" color="primary">
+                  Start the game
```
Comment (AdiYElisha) [inline frontend/src/screens/WaitingRoom.js:37]: start is sufficient

File: frontend/package.json [hunk 3]
```diff
@@ -7,8 +7,10 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
+    "axios": "^0.19.2",
     "react": "^16.13.1",
-    "react-dom": "^16.13.1"    
+    "react-dom": "^16.13.1",
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/package.json:12]: Why not? what is the problem?

File: backend/api_server.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
 // import express (after npm install express)
 const express = require("express");
+var bodyParser = require("body-parser");
```
[the remark this answers is not in the document]
Comment (aviadch) [reply backend/api_server.js]: https://github.com/aviadch/Mafia/pull/5/commits/c9dabb1b5b2c94578f1fb4bf638b83489488f7dc

File: backend/lib/server_code/api/roomRoutes.js [hunk 5]
```diff
@@ -9,11 +10,12 @@ let roomPlayers = [];
 roomRouter.use((req, res, next) => {
   res.date = Date.now();
   console.log("Time: ", Date.now());
+
```
[the remark this answers is not in the document]
Comment (aviadch) [reply backend/lib/server_code/api/roomRoutes.js]: https://github.com/aviadch/Mafia/pull/5/commits/c9dabb1b5b2c94578f1fb4bf638b83489488f7dc

File: frontend/src/components/EnterName.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import React, { Component } from "react";
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/components/EnterName.js]: https://github.com/aviadch/Mafia/pull/5/commits/7aaf845ac5752c037ddb376e369d4ec5c7309f93

File: frontend/src/components/EnterName.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  handleOnChange = event => {
+    this.setState({ name: event.target.value });
+  };
+
+  render() {
+    return (
+      <MyContext.Consumer>
+        {context => (
+          <>
+            <div className="Enter Name">
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/components/EnterName.js]: https://github.com/aviadch/Mafia/pull/5/commits/6d896d9a5c3e303c90edff346eb03042021fa84f

File: frontend/src/infra/MyProvider.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    currentRoom: "",
+    isUserEntered: false,
+    playerName: "",
+    playerId: "",
+    playerList: [],
+    joinDate: "",
+    roomCreationDate: ""
+  };
+
+  onNewGame = () => {
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/infra/MyProvider.js:19]: Iam gonna cahnge all the context api we use in the app, then will put all handelrs outside

File: frontend/src/infra/MyProvider.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    const playerId = Shortid.generate();
+    this.setState({
+      phase: PHASE.WAITING_ROOM,
+      playerId: playerId
+    });
+    const req = { creatorID: playerId };
+    axios.post(SERVER_ADDRESS + "/room/create", req).then(res => {
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/infra/MyProvider.js]: https://github.com/aviadch/Mafia/pull/5/commits/3cd00ede897f2c74f75ca5fce2cc32986bfcfb96

File: frontend/src/infra/MyProvider.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        currentRoom: roomID,
+        roomCreationDate: creationDate
+      });
+    });
+  };
+
+  onNameEntered = name => {
+    axios
+      .get(
+        SERVER_ADDRESS +
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/infra/MyProvider.js]: https://github.com/aviadch/Mafia/pull/5/commits/3cd00ede897f2c74f75ca5fce2cc32986bfcfb96

File: frontend/src/infra/MyProvider.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+          "/room/join?userID=" +
+          this.state.playerId +
+          "&roomID=" +
+          this.state.currentRoom
+      )
+      .then(res => {
+        const { joinDate, roomPlayers, error, errorMessage } = res.data;
+        if (error) {
+          console.log(errorMessage);
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/infra/MyProvider.js:43]: I know... but for now its good

File: frontend/src/screens/WaitingRoom.js (hunk 2 above)
Comment (aviadch) [reply frontend/src/screens/WaitingRoom.js]: https://github.com/aviadch/Mafia/pull/5/commits/f489d1aafffdaa14006e57949f8ef33fccdc3b97

File: frontend/src/screens/WaitingRoom.js (hunk 1 above)
Comment (aviadch) [reply frontend/src/screens/WaitingRoom.js:13]: Yes. 
When user enter the name its change and the page with the InputName is not longer needed

**Pull request merged**